modules/extensions/src/backend/index.ts

```diff
@@ -28,10 +33,97 @@ const onServerReady = async (bp: typeof sdk) => {
       res.sendStatus(404)
     })
   )
+
+  await setupMiddleware(bp)
+}
+
+const onModuleUnmount = async (bp: typeof sdk) => {
+  await removeMiddleware(bp)
+}
+
+async function setupMiddleware(bp: typeof sdk) {
+  bp.events.registerMiddleware({
+    description: 'Extends outgoing messages',
+    direction: 'outgoing',
+    handler: outgoingHandler,
+    name: MIDDLEWARE_NAME,
+    order: 10
+  })
+
+  async function outgoingHandler(event: sdk.IO.OutgoingEvent, next: sdk.IO.MiddlewareNextCallback) {
+    const logger = bp.logger.forBot(event.botId)
+    // logger.info(`outgoingHandler: ${JSON.stringify(event)}`)
+
+    if (event.payload && event.payload.options) {
+      events[event.target] = event
+    }
+
+    return next()
+  }
+
+  bp.events.registerMiddleware({
+    description: 'Extends incoming messages',
+    direction: 'incoming',
+    handler: incomingHandler,
+    name: MIDDLEWARE_NAME,
+    order: 0
+  })
+
+  function normalize(str: string) {
+    return str
+      .toLowerCase()
+      .normalize('NFD')
+      .replace(/[\u0300-\u036f]/g, '')
+      .split(/ \(|,/)[0]
+  }
+
+  async function incomingHandler(event: sdk.IO.IncomingEvent, next: sdk.IO.MiddlewareNextCallback) {
+    const logger = bp.logger.forBot(event.botId)
+    const outgoingEvent = events[event.target]
```

Comment:
This would not work in a multi node environment

Reply:
I noticed this module exposes an API to retrieve events from the db. Would that be viable alternative?